src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs

```diff
@@ -71,13 +72,14 @@ private void SetupResolver(MessagePackHubProtocolOptions options)
                 // check if the user customized the resolvers
                 if (options.FormatterResolvers[i] != SignalRResolver.Resolvers[i])
                 {
-                    _resolver = new CombinedResolvers(options.FormatterResolvers);
+                    var resolver = new CombinedResolvers(options.FormatterResolvers);
```

Comment:
I take it `CombinedResolvers` is a private type. Consider removing that type and using the new [`CompositeResolver`](https://github.com/neuecc/MessagePack-CSharp/#create-own-custom-composite-resolver) instead.

Reply:
done. (no alternative for the `.ToArray()` i guess ?

Reply:
You can call the overload that takes a `IReadOnlyList` to avoid the `ToArray()` call yourself.
Either way though, `ToArray()` will be called internally. But if you're concerned about efficiency here, consider storing your resolver and reusing it because the `CompositeResolver` has a cache within itself that will have some cost to build up but then improves throughput perf.

Reply:
Yep, i'll use an `IReadOnlyList` cast, since as said, [`ToArray()` is called internally](https://github.com/neuecc/MessagePack-CSharp/blob/master/src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Resolvers/CompositeResolver.cs#L52-L53)

Regarding the possible cached value, the `SetupResolver` is only [referenced and called one in the `MessagePackHubProtocol` constructor](https://github.com/dotnet/aspnetcore/blob/master/src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs#L53-L57) I'm currently crawling all places that could resolve the `IHubProtocol` to know how it could be done. I'm not confident enough to "just change it" right now ;)